src/core/credentials/transport/tls/spiffe_utils.cc

```diff
@@ -156,4 +167,114 @@ absl::StatusOr<SpiffeId> SpiffeId::FromString(absl::string_view input) {
   return SpiffeId(trust_domain, absl::StrCat("/", path));
 }
 
+void SpiffeBundle::JsonPostLoad(const Json&, const JsonArgs&,
+                                ValidationErrors* errors) {
+  {
+    ValidationErrors::ScopedField field(errors, ".keys");
+    for (size_t i = 0; i < keys_.size(); i++) {
+      auto root = keys_[i].GetRoot();
+      if (!root.ok()) {
+        errors->AddError(absl::StrFormat("Cannot get root certificate: %s",
+                                         root.status().ToString()));
+      } else {
+        roots_.push_back(*root);
+      }
+    }
+  }
+}
+
+void SpiffeBundleKey::JsonPostLoad(const Json&, const JsonArgs&,
+                                   ValidationErrors* errors) {
+  {
+    ValidationErrors::ScopedField field(errors, ".use");
+    if (use_ != kAllowedUse) {
+      errors->AddError(
+          absl::StrFormat("got %s. Only supported value for use field is %s.",
+                          use_, kAllowedUse));
+      return;
+    }
+  }
+  {
+    ValidationErrors::ScopedField field(errors, ".kty");
+    if (kty_ != kAllowedKty) {
+      errors->AddError(
+          absl::StrFormat("got %s. Only supported value for kty field is %s",
+                          kty_, kAllowedKty));
+      return;
+    }
+  }
+  {
+    ValidationErrors::ScopedField field(errors, ".x5c");
+    if (x5c_.size() != kX5cSize) {
+      errors->AddError(absl::StrFormat(
+          "got vector length %i. Expected length of exactly %i.", x5c_.size(),
+          kX5cSize));
+      return;
+    }
+
+    std::string pem_cert =
+        absl::StrCat(kCertificatePrefix, x5c_[0].data(), kCertificateSuffix);
+    auto certs = ParsePemCertificateChain(pem_cert);
+    if (!certs.ok()) {
+      errors->AddError(certs.status().ToString());
+      return;
+    }
+    for (X509* cert : *certs) {
+      X509_free(cert);
+    }
+  }
+}
+
+void SpiffeBundleMap::JsonPostLoad(const Json&, const JsonArgs&,
+                                   ValidationErrors* errors) {
+  {
+    ValidationErrors::ScopedField field(errors, ".trust_domains");
```

Comment:
Suggest writing this as:

```
for (auto const& [k, _] : temp_bundles_) {
  ValidationErrors::ScopedField field(
      errors, absl::StrCat(".trust_domains[\"", k, "\"]"));
  absl::Status status = ValidateTrustDomain(k);
  if (!status.ok()) {
    errors->AddError(absl::StrCat("invalid trust domain: ", status.ToString()));
  }
}
```

This approach reports the map key as part of the field name instead of the error message.  It also uses C++14 structured bindings to improve readability.

Reply:
Done, thanks for the suggestion!